src/db/transaction.cr

```diff
@@ -0,0 +1,118 @@
+module DB
+  abstract class Transaction
+    include Disposable
+    include BeginTransaction
+
+    abstract def connection : Connection
+
+    def commit
+      close!
+    end
+
+    def rollback
+      close!
+    end
+
+    private def close!
+      raise DB::Error.new("Transaction already closed") if closed?
```

Comment:
I'm wondering if we maybe should let just the DB fail in this case, rather than try to play smarter and throw the exception ourselves. However, I can't think of any scenario where we think the tx is `closed?` but it actually isn't.